File: Front/ProEventos-app/src/app/interfaces/models/palestrantes/Palestrante.ts
import { RedeSocial } from "../redesSociais/RedeSocial";
import { UsuarioUpdate } from "../Usuarios";
import { PalestranteEvento } from "./PalestranteEvento";

export interface Palestrante {
  id: number;
  miniCurriculo: string;
  userId: string;
  user: UsuarioUpdate;
  redesSociais?: RedeSocial[];
  palestrantesEventos?: PalestranteEvento[];
}

File: Back/ProEventos.Domain/Models/Palestrantes/Palestrante.cs
using ProEventos.Domain.Models.RedesSociais;
using ProEventos.Domain.Models.Usuarios;

namespace ProEventos.Domain.Models.Palestrantes
{
    public class Palestrante
    {
        public int Id { get; set; }
        public string MiniCurriculo { get; set; }
        public int UserId { get; set; }
        public Usuario Usuario { get; set; }
        public IEnumerable<RedeSocial> RedesSociais { get; set; }
        public IEnumerable<PalestranteEvento> PalestrantesEventos { get; set; }
    }
}
